File: ast/statement/block.go
package statement

import (
	"github.com/panda-io/micro-panda/ast/ast"
)

type Block struct {
	StatementBase
	Statements []ast.Statement
}

func (b *Block) Validate(c ast.Context) {
	//TO-DO warning: unreachable code //Start, End of block
	for _, statement := range b.Statements {
		ctx := c
		if _, ok := statement.(*Block); ok {
			ctx = c.NewContext()
		}
		statement.Validate(ctx)
	}
}

File: app/src/main/java/com/github/panda_io/micro_panda/ast/statement/Block.java
package com.github.panda_io.micro_panda.ast.statement;

import java.util.*;

import com.github.panda_io.micro_panda.ast.Context;

public class Block extends Statement {
    public List<Statement> statements;
    
    //TO-DO warning: unreachable code //Start, End of block
    public void validate(Context context) {
        context = context.newContext();
        for (Statement statement:this.statements) {
            statement.validate(context);
        }
    }
}
